src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx

```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
 }
 
 export function JoinedIndexProgress(props: JoinedIndexProgressProps) {
-    const { index } = props;
-
-    if (index.nodesInfo.some((x) => x.status === "failure")) {
-        return (
-            <ProgressCircle inline state="failed" icon="cancel">
-                Load error
-            </ProgressCircle>
-        );
-    }
-    if (index.nodesInfo.some((x) => x.details?.faulty)) {
-        return (
-            <ProgressCircle inline state="failed" icon="cancel">
-                Faulty
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.state === "Error")) {
-        return (
-            <ProgressCircle inline state="failed" icon="cancel">
-                Error
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Disabled")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Disabled")}>
-                Disabled
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Paused")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Paused")}>
-                Paused
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.status === "Pending")) {
-        return (
-            <ProgressCircle inline state="running" icon={iconForState("Pending")}>
-                Pending
-            </ProgressCircle>
-        );
-    }
-
-    if (index.nodesInfo.some((x) => x.details?.stale)) {
-        const overallProgress = calculateOverallProgress(index);
-
-        return (
-            <ProgressCircle inline state="running" progress={overallProgress}>
-                Running
-            </ProgressCircle>
-        );
-    }
+    const { index, onClick } = props;
+    const getState = (index: IndexSharedInfo) => {
```

Comment:
if this is pure function move it outside

Reply:
Done